packages/lit-html/src/lit-html.ts

```diff
@@ -757,6 +840,20 @@ export class NodePart {
   private _commitNode(value: Node): void {
     if (this._value !== value) {
       this._clear();
+      const parent = this._startNode.parentNode;
+      if (
+        ENABLE_EXTRA_SECURITY_HOOKS &&
+        sanitizerFactoryInternal !== noopSanitizer &&
+        (parent?.nodeName === 'STYLE' || parent?.nodeName === 'SCRIPT')
+      ) {
+        this._insert(
+          new Text(
```

Comment:
I vaguely remember discussing allowing TemplateResults, since they're presumed safe, just not other values.

And how does this work with safe value types? Should those be allowed?

Reply:
TemplateResults aren't presumed safe in all contexts. For example, this is a template result, but isn't safe to interpolate inside a style or script tag:

```typescript
html`${somethingFromUserInput}`
```

I believe you can commit SafeScript text as the text content of a `<script>` tag, and SafeStyle as the text content of a `<style>` tag, but in order to keep complexity down we're just banning rendering anything more complex inside a `<script>` or `<style>`

Reply:
Won't that template result have any bindings within it checked by another call to `_commitNode`?

Reply:
Hm, maybe? I think this option is safe and simple though, and we haven't had demand for the ability to write into scripts and style tags. Expanding options is back-compatible, contracting them isn't. Let's start with this.

Reply:
sounds good